### PR TITLE
feature/TECH 561 gha integration test

### PR DESCRIPTION
**Pull request by SamTheisens**
Clones out repo in minimalist way, only including differences from main.

Screenshot 2023-09-09 at 17 26 47

branch: feature/TECH-561-test-mpyl-in-gha

----
### 📕 [TECH-561](https://vandebron.atlassian.net/browse/TECH-561) Create regression test for tag builds sam@vandebron.nl 
Create a github action for a PR that runs in the mpyl repo on master with the test version of the PR.

This is to ensure that the tag flow doesn't break with every merge.

Ensure that the jenkinsfile in the mpyl repo is the same as the monorepo one (i.e. same mpyl commands are run in the same order).

🏗️ Build [4](https://jenkins.k8s-serv-backend.vdbinfra.nl/job/MPyL%20Pipeline%20-%20Test/job/PR-233/4/display/redirect) ✅ Successful, started by _Sam Theisens_  
🚀 *[cloudfront-service](https://cloudfront-service-233.test.nl/)*, *example-dagster-user-code*, *job*, *kong-sync*, *[nodeservice](https://nodeservice-233.test.nl/)*, *[sbtservice](https://sbtservice-233.test.nl/)*, *sparkJob*  

[TECH-561]: https://vandebron.atlassian.net/browse/TECH-561?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/build-package.yml [hunk 1]
```diff
@@ -162,10 +172,110 @@ jobs:
...
+      - name: Merge branch
+        run: git merge --no-ff -m "Merge PR ${{ github.ref }}" ${{ github.ref }}
+
+      - name: Tag merge
+        run: git tag -a v${{ needs.Build_Package.outputs.mpylVersion }} -m "Tagging v${{ needs.Build_Package.outputs.mpylVersion }}"
+
+      - name: Show build status
+        env:
+          TAG_NAME: v${{ needs.Build_Package.outputs.mpylVersion }}
+        run: mpyl build status
```
Comment (Jorg88) [inline .github/workflows/build-package.yml:281]: not running the build?

Comment (SamTheisens) [reply .github/workflows/build-package.yml:281]: We could, but not sure how much the added value is. Besides the change detection mechanism there's no difference with PR builds.

File: .github/workflows/build-package.yml [hunk 2]
```diff
@@ -162,10 +172,110 @@ jobs:
...
+        run: mpyl repo status
+
+      - name: Check mpyl health
+        run: mpyl health --ci
+
+      - name: Lint projects
+        run: mpyl projects lint
+
+      - name: Validate upgrade status
+        run: mpyl projects upgrade
```
[the remark this answers is not in the document]
Comment (SamTheisens) [reply .github/workflows/build-package.yml:216]: It gives a suggestion to run lint after you've done upgrade

File: .github/workflows/build-package.yml (hunk 1 above)
Comment (SamTheisens) [reply .github/workflows/build-package.yml:289]: https://github.com/Vandebron/mpyl/pull/233/commits/ce747c7c42a25a7c12e44e37a47a3fb09e9b6b51

**Pull request merged**